### ISSUE TITLE
[DICT] Add Python support for update

### ISSUE THREAD

**Issue by EmilyBourne** [pyccel/pyccel#1889]
### Describe the feature
Syntactic and semantic support should be added for `update`. This should be tested by implementing Python printing

### Test Code
Provide code which does not currently work but which should do when this issue is fixed:
```python
def foo():
    a = {1:1.0, 2:2.0}
    b = {3:3.0, 1:4.0}
    a.update(b)
    return a, b
```

**Comment by learning-dev** [comment]
Hi @EmilyBourne, can you please assign this to me?

**Comment by EmilyBourne** [comment]
@learning-dev now #2033 has been successfully merged are you still interested in working on this issue? Do you want me to assign it to you?

**Comment by learning-dev** [comment]
@EmilyBourne  I was thinking of an issue in 'C'  language. But anyways, you can assign this to me. I will take this up.

**Comment by EmilyBourne** [comment]
#1948 is already being fixed by @jalalium. I need to open some issues for the C translations of the dict methods though if you are interested?
Or you can do this one and include the C support

**Comment by learning-dev** [comment]
@EmilyBourne, yeah I can take this up with C support. Apart from printing in C. What else should I add in C?

**Comment by EmilyBourne** [comment]
I don't think there's a direct translation to C for this method in STC so you might need to add a dedicated function. I did C support for some of the set methods last week there are probably some examples there. I'll send you some links tomorrow

**Comment by learning-dev** [comment]
Sure, thank you.

**Comment by EmilyBourne** [comment]
Hello again,

All the developer docs for Pyccel can be found in this folder:
https://github.com/pyccel/pyccel/tree/devel/developer_docs

The STC docs describing the map methods are found here:
https://github.com/stclib/STC/blob/master/docs/hmap_api.md

There are multiple ways to handle `update`. Either you can interpret it as multiple calls to `setdefault` and translate:
```python
def foo():
    a = {1:1.0, 2:2.0}
    b = {3:3.0, 1:4.0}
    a.update(b)
    return a, b
```
to
```python
def foo():
    a = {1:1.0, 2:2.0}
    b = {3:3.0, 1:4.0}
    for b_k, b_v in b:
        a.setdefault(b_k, b_v)
    return a, b
```
or you can create a dedicated function in C.

The first option is very similar to what is done for `list.extend` (see #1731) and for `set.update` (see #1843). However issue #1884 (or #1882 and #1883) would need to be fixed first.

The second option is similar to what is done for `set.union` (see #2015).

Another thing to beware of is that there is no support for returning dictionaries from C yet so you will need to return other things in order to test the result. For example you can use `len` normally `pop` or `popitem` would also be useful, but we don't have C support for those yet.

It is up to you. Maybe it is better if I open the C issues for pop/popitem and you just tackle the Python version for now to avoid tripping over these issues while I try and get the C-Python API wrapper updated so we can return container objects

**Comment by learning-dev** [comment]
Hi @EmilyBourne, I'm overwhelmed with so many details. I will start with the python bit first and then attempt the C part. I attempted to read the code and I understood a lot of pointers  ( pass by pointers) are being used. 

Hopefully, I will be done by this weekend thanks.

**Comment by EmilyBourne** [comment]
@learning-dev sorry for the information overload :sweat_smile:
Maybe this isn't the simplest issue right now.
I could recommend #1744 as an issue that doesn't have so many options

**Comment by learning-dev** [comment]
@EmilyBourne  can you please assign #1744  to me and unassign this?
